consensus/poet/core/tests/test_consensus/test_poet_fork_resolver.py

```diff
@@ -0,0 +1,649 @@
+    import ValidatorInfo
+from sawtooth_poet_common.protobuf.validator_registry_pb2 \
+    import SignUpInfo
+
+
+@mock.patch('sawtooth_poet.poet_consensus.poet_block_publisher.BlockWrapper')
+@mock.patch('sawtooth_poet.poet_consensus.poet_block_publisher.PoetConfigView')
+class TestPoetBlockResolver(TestCase):
+
+    def setUp(self):
+        # pylint: disable=invalid-name,global-statement
+        global poet_fork_resolver
+        # PoetBLockResolver uses class variables to hold state
+        # so the module needs to be reloaded after each test to clear state
+        poet_fork_resolver = reload(poet_fork_resolver)
+
+        self._temp_dir = tempfile.mkdtemp()
+
+    def tearDown(self):
+        shutil.rmtree(self._temp_dir)
+
+    @mock.patch('sawtooth_poet.poet_consensus.poet_fork_resolver.'
+                'ConsensusStateStore')
+    @mock.patch('sawtooth_poet.poet_consensus.poet_fork_resolver.factory')
+    @mock.patch('sawtooth_poet.poet_consensus.poet_fork_resolver.'
+                'ConsensusState')
+    @mock.patch('sawtooth_poet.poet_consensus.poet_fork_resolver.'
+                'ValidatorRegistryView')
+    @mock.patch('sawtooth_poet.poet_consensus.poet_fork_resolver.utils')
+    def test_new_fork_head_not_valid_block(
+            self,
+            mock_utils,
+            mock_validator_registry_view,
+            mock_consensus_state,
+            mock_poet_enclave_factory,
+            mock_consensus_state_store,
+            mock_poet_config_view,
+            mock_block_wrapper):
+
+        """ Test verifies that if the new fork head is not a valid block,
+            exits
+        """
+
+        # create a mock_validator_registry_view
+        mock_validator_registry_view.return_value.get_validator_info. \
+            return_value = \
+            ValidatorInfo(
+                name='validator_001',
+                id='validator_deadbeef',
+                signup_info=SignUpInfo(
+                    poet_public_key='00112233445566778899aabbccddeeff'))
+
+        # Make utils pretend it cannot deserialize the wait certificate
+        # of the new fork head
+        mock_utils.deserialize_wait_certificate.return_value = None
+
+        # create a mock_consensus_state that returns a mock with
+        # the following settings:
+        mock_state = mock.Mock()
+        mock_state.validator_signup_was_committed_too_late.return_value = False
+        mock_state.validator_has_claimed_block_limit.return_value = False
+        mock_state.validator_is_claiming_too_early.return_value = False
+        mock_state.validator_is_claiming_too_frequently.return_value = False
+
+        mock_block_cache = mock.MagicMock()
+        mock_state_view_factory = mock.Mock()
+
+        # create mock_poet_enclave_module
+        mock_poet_enclave_module = mock.Mock()
+        mock_poet_enclave_module.return_value = \
+            mock_poet_enclave_factory.get_poet_enclave_module.return_value
```

Comment:
If you mean for `mock_poet_enclave_module` to be returned by the factory, you need this to be:
`mock_poet_enclave_factory.get_poet_enclave_module.return_value = mock_poet_enclave_module`